src/pages/groupComparison/Survival.tsx

```diff
@@ -26,12 +27,31 @@ export default class Survival extends React.Component<ISurvivalProps, {}> {
             this.props.store.uidToGroup
         ],
         invoke: () => {
+            const orderedActiveGroupUidSet = _.reduce(this.props.store._activeGroupsNotOverlapRemoved.result!, (acc, next, index) => {
+                acc[next.uid] = index;
+                return acc;
+            }, {} as { [id: string]: number });
             const partition = this.props.store.patientsVennPartition.result!;
+
+            // ascending sort partition bases on number of groups in each parition.
+            // if they are equal then sort based on the give order of groups
+            partition.sort((a, b) => {
+                const aUids = Object.keys(a.key).filter(uid=>a.key[uid]);
```

Comment:
i don't quite understand this.  you're taking keys of aj object (a.key) then filtering out those where the corresponding values is false?    seems like it would be a lot clearer with reduce

Reply:
@alisman I asked Karthik to change it from reduce. I don't see what is unclear about this, it's a one liner, it does what it says. why would reduce make it more clear?